feat: enhance SidePanel and TestCase management with new components and API integration

MahirSalahin opened the pull request:
- Added DeleteDialog component for confirmation of deleting testcase.
- Introduced TestCaseForm component for creating and editing test cases.
- Updated TestCaseList to support creating, editing, and deleting test cases with appropriate UI changes.
- Implemented API methods for creating, updating, and deleting test cases.
- Enhanced SidePanel with dismiss functionality and improved UI elements.

MahirSalahin committed:
feat: enhance SidePanel and TestCase management with new components and API integration

- Added AlertDialog component for confirmation dialogs.
- Introduced TestCaseForm component for creating and editing test cases.
- Updated TestCaseList to support creating, editing, and deleting test cases with appropriate UI changes.
- Implemented API methods for creating, updating, and deleting test cases.
- Enhanced SidePanel with dismiss functionality and improved UI elements.

adnantech17 commented on pages/side-panel/src/components/AlertDialog.tsx:
```diff
@@ -0,0 +1,102 @@
+  isOpen,
+  title,
+  message,
+  confirmText = 'Delete',
```
the default value of a confirm button for an Alert Dialog should be something like 'Ok' or 'Submit'. Delete doesn't make sense

MahirSalahin replied:
> the default value of a confirm button for an Alert Dialog should be something like 'Ok' or 'Submit'. Delete doesn't make sense

Actually it is a Delete Dialog, renamed it to DeleteDialog.

adnantech17 commented on pages/side-panel/src/components/TestCaseForm.tsx:
```diff
@@ -0,0 +1,219 @@
+
+interface TestCaseFormProps {
+  projectId: string;
+  testCase?: TestCase; // If provided, we're editing an existing test case
```
remove comment

adnantech17 commented on pages/side-panel/src/components/TestCaseForm.tsx:
```diff
@@ -0,0 +1,219 @@
+  }, [testCase]);
+
+  const handleAddStep = () => {
+    setSteps([...steps, '']);
```
setSteps(prevSteps => [...prevSteps, ''])

adnantech17 commented on pages/side-panel/src/components/TestCaseForm.tsx:
```diff
@@ -0,0 +1,219 @@
+
+  const handleRemoveStep = (index: number) => {
+    if (steps.length > 1) {
+      setSteps(steps.filter((_, i) => i !== index));
```
prevSteps => prevSteps.filter()

adnantech17 commented on pages/side-panel/src/components/TestCaseList.tsx:
```diff
@@ -58,20 +71,116 @@ export default function TestCaseList({ projectId, onTestCaseSelect, onBackToProj
+  };
+
+  const handleEditClick = (testCase: TestCase, e: React.MouseEvent) => {
+    e.stopPropagation(); // Prevent triggering the test case selection
```
remove comments. same for all

adnantech17 commented on pages/side-panel/src/components/TestCaseList.tsx:
```diff
@@ -58,20 +71,116 @@ export default function TestCaseList({ projectId, onTestCaseSelect, onBackToProj
+      const errorMessage = err instanceof Error ? err.message : 'Unknown error';
+      setDeleteError(`Failed to delete test case: ${errorMessage}`);
+      // Clear error after 5 seconds
+      setTimeout(() => setDeleteError(null), 5000);
```
why?

MahirSalahin replied:
> why?

removed

adnantech17 commented on pages/side-panel/src/services/api.ts:
```diff
@@ -73,6 +73,49 @@ export class ApiClient {
+  async put<T>(path: string, data: unknown): Promise<T> {
+    try {
+      const headers = await this.getHeaders();
+      const response = await fetch(`${this.baseUrl}${path}`, {
```
aren't we using axios?

MahirSalahin replied:
> aren't we using axios?

Using now.

adnantech17 commented on pages/side-panel/src/services/api.ts:
```diff
@@ -73,6 +73,49 @@ export class ApiClient {
+      });
+
+      if (!response.ok) {
+        if (response.status === 401) {
```
use axios and handle these in interceptor?

MahirSalahin committed:
feat: update SidePanel and enhance test case management

- Refactored SidePanel to include new state management for test cases and improved UI interactions.
- Added DeleteDialog component for confirming deletions of test cases.
- Updated TestCaseList to utilize DeleteDialog and manage test case creation and editing more effectively.
- Enhanced API client to streamline request handling and error management.
- Improved overall code structure and readability across components.
